src/background.js

```diff
@@ -4,40 +4,68 @@ import he from "he";
 
 const BOOKMARK_BAR_ID = "1";
 
+/**
+ * Storage change 
+ * @typedef {Object} StorageChange
+ * @property {import("./config.js").FeedConfig} [newValue]
+ * @property {import("./config.js").FeedConfig} [oldValue]
+ */
+
+/**
+ * Initializes
+ * @returns {Promise<void>} 
+ */
 async function init() {
   const config = await loadConfig();
 
   console.log("using configuration", JSON.stringify(config));
-  updateFeeds(config);
+  await updateFeeds(config);
 }
 
-//polls each configured feed with a delay between each to reduce load all at once
+/**
+ * Polls each configured feed with a delay between each to reduce load all at once
+ * @param {import("./config.js").FeedConfig} config
+ * @returns {Promise<void>} 
+ */
 async function updateFeeds(config) {
-  config.feeds.forEach(updateFeedBookmarks);
+  for (const feed of config.feeds) {

```

Comment:
forEach is not designed to work with with async code, we should use for of. More info: https://gist.github.com/joeytwiddle/37d2085425c049629b80956d3c618971